Base/QTApp/qSlicerApplicationHelper.cxx

```diff
@@ -106,6 +106,7 @@ void qSlicerApplicationHelper::preInitializeApplication(
   // support volume rendering and other VTK functionality that reads from the
   // framebuffer; see https://gitlab.kitware.com/vtk/vtk/issues/17095.
   QSurfaceFormat format = QVTKOpenGLWidget::defaultFormat();
+  format.setProfile(QSurfaceFormat::CompatibilityProfile);
```

Comment:
Could this be added into its own comment along with referencing the discussion on VTK discourse ?

It looks we should introduce an helper function and update all tests to use that. See https://github.com/Slicer/Slicer/blob/2f8f88fa961375b5c902fdc7ec5ca9e6d5200640/Base/QTGUI/Testing/Cxx/qSlicerMouseModeToolBarTest1.cxx#L42-L47

I agree that ideally we should improve VTK to initialize the surface appropriately but that is its own "can of worms".  to move forward, adding an helper function allowing to perform initialization for all tests would make sense i think.